toolchain/check/context.cpp

```diff
@@ -187,6 +187,82 @@ auto Context::DiagnoseNameNotFound(SemIRLoc loc, SemIR::NameId name_id)
   emitter_->Emit(loc, NameNotFound, name_id);
 }
 
+auto Context::DiagnoseInvalidAccess(
+    Parse::NodeId node_id, SemIR::InstId member_id, SemIR::NameId name_id,
+    SemIR::AccessKind access_kind, bool is_parent_access,
+    std::optional<AccessInfo> access_info) -> void {
+  if (!access_info) {
+    return;
+  }
+
+  auto inst = insts().Get(access_info->base_id);
+  auto inst_type_id = inst.type_id();
+
+  if (auto class_type = types().TryGetAs<SemIR::ClassType>(inst_type_id)) {
```

Comment:
```suggestion
  // TODO: Support scoped entities other than just classes.
  if (auto class_type = types().TryGetAs<SemIR::ClassType>(inst_type_id)) {
```

I assume you're deliberately not supporting other entities, but this should be called out as a TODO.